llvm/test/CodeGen/SPIRV/duplicate-types.ll

```diff
@@ -0,0 +1,16 @@
+; Check that we don't end up with duplicated array types in TypeMap.
+; No FileCheck needed, we only want to check the absence of errors.
+; RUN: llc -O0 -mtriple=spirv64-unknown-unknown %s -o -
```

Comment:
I'd suggest running `-verify-machineinstrs` as we do with other intrinsic tests. I believe this would still be verified on build bots running tests with expensive checks enabled, but we can always run verification in regular build as well.
```suggestion
; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv64-unknown-unknown %s -o -
```